### PR TITLE
Decoupler module

### LINKED ISSUE #5098: new module: decoupleR
### Is there an existing module for this?

- [x] I have searched for the existing module

### Is there an open PR for this?

- [x] I have searched for existing PRs

### Is there an open issue for this?

- [x] I have searched for existing issues

### Are you going to work on this?

- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by Nic-Nic**
## PR checklist

Closes #5098 

Adds nf-core module for running decoupler-py, a python package suite for omic data enrichment analysis

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR. -> [Here](https://github.com/nf-core/test-datasets/pull/1137)
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by kbestak — CHANGES_REQUESTED** [review]

File: tests/config/pytest_modules.yml
[the remark this answers is not in the document]
Comment (vicpaton) [reply tests/config/pytest_modules.yml:1]: it was triggering an error in the linting

File: .gitignore
[the remark this answers is not in the document]
Comment (vicpaton) [reply .gitignore:1]: yes, we can discard them. We were running the process for debugging and we did not want the output files to be pushed to the repo

File: modules/nf-core/decoupler/tests/nextflow.config
[the remark this answers is not in the document]
Comment (vicpaton) [reply modules/nf-core/decoupler/tests/nextflow.config:1]: you mean the enabled=true? i think we can remove this

[the remark this answers is not in the document]
Comment (kbestak) [reply modules/nf-core/decoupler/tests/nextflow.config:1]: The whole docker profile block

[the remark this answers is not in the document]
Comment (vicpaton) [reply modules/nf-core/decoupler/tests/nextflow.config:1]: empirically, yes ahahah. I don't know why though

File: modules/nf-core/decoupler/main.nf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    tuple val(meta), path("*estimate__decoupler.tsv"), emit: dc_estimate
+    tuple val(meta), path("*pvals__decoupler.tsv"), emit: dc_pvals
+    path("versions.yml"), emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: '{}'
+    def methods = task.ext.methods ?: 'None'
```
[the remark this answers is not in the document]
Comment (vicpaton) [reply modules/nf-core/decoupler/main.nf]: adressed

File: modules/nf-core/decoupler/tests/nextflow.config
[the remark this answers is not in the document]
Comment (Nic-Nic) [reply modules/nf-core/decoupler/tests/nextflow.config:1]: Fixed in a8eace7

**Review by kbestak — APPROVED** [review]
LGTM!
Great work

**Pull request merged**